Page-specific Banner

MalcolmVonMoJ opened the pull request:
## What does this pull request do?

Adds in a page-specific banner option via ACF
- A text field
- a series of links (max number configurable per post type, default = 2)
- whether the banner is enabled for the post type (default = no)
- whether the banner is shown on this specific post (default = no)

## What is the new Hale version number?

4.19.0

## Is the change available on the Dev or Demo environments?

Dev

## Checklist

- [x] Checked on a mobile device
- [x] Checked on a desktop device
- [x] Checked on Safari
- [ ] Checked on Firefox
- [x] Checked on Chrome

## Notes

brown-a2 commented on partials/information-banner.php:
```diff
@@ -39,3 +39,38 @@
+    // Main text
+    $banner_text = get_field('post_banner_text');
+    if ($banner_text && $banner_text != "") {
+        $banner_content .= "<p class='govuk-body page-banner__text'>$banner_text</p>";
```
As this is banner text do we need to escape i.e, esc_html( $banner_text) ?

MalcolmVonMoJ replied:
Done

brown-a2 commented on partials/information-banner.php:
```diff
@@ -39,3 +39,38 @@
+    // Get max number of links
+    $number_of_banner_links = hale_get_post_type_setting('single_view_banner_max_links');
+
+    if (!$number_of_banner_links) $number_of_banner_links = 0;
```
If you want to make clear it is evaluating a boolean, you could do it like `$number_of_banner_links = ( isset( $number_of_banner_links ) && $number_of_banner_links ) ? $number_of_banner_links : 0;` Although I appreciate it isn't as concise.

MalcolmVonMoJ replied:
Done

brown-a2 commented on partials/information-banner.php:
```diff
@@ -39,3 +39,38 @@
+        if (!is_array($link)) continue;
+
+        $link_text = $link["title"];
+        $link_url = $link["url"];
```
Escaping
```
$link_text = esc_html($link_text);  
$link_url = esc_url($link_url);
```

MalcolmVonMoJ replied:
Done

brown-a2 commented on template-parts/content-none.php:
```diff
@@ -12,9 +12,9 @@
 ?>
 
 <section class="no-results not-found">
-	<header class="entry-header">
```
👍

brown-a2 reviewed (commented):
To other observations 
- When I was viewing in on mobile (on my brower inspect mobile) the margins/padding didn't seem to be there. The banner text was right up against the edge of the screen.
- Because the banner is appearing only in the single view, should it be in the Display Settings area in the backend rather then its own area?

MalcolmVonMoJ commented on inc/acf/post-type-fields/field-banner.php:
```diff
@@ -0,0 +1,115 @@
+            ]);
+
+            if ($number_of_banner_links) $input_width = 100/$number_of_banner_links . "%"; //up to 6 choices, we lay them out side-by-side,
+
```
Done

MalcolmVonMoJ commented:
I have moved the banner controls to the display settings tab.  

I have added a class to ensure the width is the same as the content on mobile.  Good spot!

brown-a2 commented on inc/acf/admin/post-display-settings.php:
```diff
@@ -9,6 +9,8 @@
 add_action('acf/post_type/render_settings_tab/display-settings', function ($acf_post_type) {
 
+    $post_label = $acf_post_type['labels']['singular_name'];
+
```
In case for some reason it can't get the label I'd write like this $post_label = $acf_post_type['labels']['singular_name'] ?? null;

MalcolmVonMoJ replied:
Done
